File: datasets/Part 1 - Data Preprocessing/missing_data.R

# Plantilla para el Pre Procesado de Datos - Datos faltantes

setwd("~/repos/machinelearning-az/datasets/Part 1 - Data Preprocessing")

# Importar el dataset
dataset = read.csv('data/Data.csv')


# Tratamiento de los valores NA
dataset$Age = ifelse(is.na(dataset$Age),
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     dataset$Age)

dataset$Salary = ifelse(is.na(dataset$Salary),
                        ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        dataset$Salary)
                        

File: datasets/Part 1 - Data Preprocessing/missing_data.py
# -*- coding: utf-8 -*-
"""
Created on Fri Apr 29 12:44:10 2022

@author: mudar
"""

# Plantilla de Pre Procesado - Datos faltantes

# Como importar las librerias
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el dataset
dataset = pd.read_csv('data/Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

# Tratamiento de los NAs
from sklearn.impute import SimpleImputer

# Sustituir los valores nan por la media de la columna
imputer = SimpleImputer(missing_values = np.nan, strategy = 'mean', verbose=0)
imputer = imputer.fit(X[:, 1:3])
X[:, 1:3] = imputer.transform(X[:, 1:3])
